src/stripe.common.ts

```diff
@@ -0,0 +1,498 @@
+import _Error = require('./Error');
+import http = require('http');
+
+import resources = require('./resources');
+
+const DEFAULT_HOST = 'api.stripe.com';
+const DEFAULT_PORT = '443';
+const DEFAULT_BASE_PATH = '/v1/';
+const DEFAULT_API_VERSION = (null as unknown) as string;
+
+const DEFAULT_TIMEOUT = 80000;
+
+Stripe.PACKAGE_VERSION = require('../package.json').version;
+
+const determineProcessUserAgentProperties = (): Record<string, string> => {
+  return typeof process === 'undefined'
+    ? {}
+    : {
+        lang_version: process.version,
+        platform: process.platform,
+      };
+};
+
+Stripe.USER_AGENT = {
+  bindings_version: Stripe.PACKAGE_VERSION,
+  lang: 'node',
+  publisher: 'stripe',
+  uname: null,
+  typescript: false,
+  ...determineProcessUserAgentProperties(),
+};
+
+/** @private */
+Stripe._UNAME_CACHE = null as Promise<string> | null;
+
+const MAX_NETWORK_RETRY_DELAY_SEC = 2;
+const INITIAL_NETWORK_RETRY_DELAY_SEC = 0.5;
+
+const APP_INFO_PROPERTIES = ['name', 'version', 'url', 'partner_id'];
+const ALLOWED_CONFIG_PROPERTIES = [
+  'apiVersion',
+  'typescript',
+  'maxNetworkRetries',
+  'httpAgent',
+  'httpClient',
+  'timeout',
+  'host',
+  'port',
+  'protocol',
+  'telemetry',
+  'appInfo',
+  'stripeAccount',
+];
+
+import StripeResource = require('./StripeResource');
+import RequestSender = require('./RequestSender');
+Stripe.StripeResource = StripeResource;
+Stripe.resources = resources;
+
+import HttpClient = require('./net/HttpClient');
+Stripe.HttpClient = HttpClient.HttpClient;
+Stripe.HttpClientResponse = HttpClient.HttpClientResponse;
+
+import CryptoProvider = require('./crypto/CryptoProvider');
+import EventEmitter = require('events');
+Stripe.CryptoProvider = CryptoProvider;
+
+import DefaultPlatformFunctions = require('./platform/DefaultPlatformFunctions');
+Stripe._platformFunctions = new DefaultPlatformFunctions();
```

Comment:
How about setting this below in prototype definition?

Reply:
Oh, this is a "global" and we are copying the value to an instance on every creation. Interesting.

Reply:
Since I updated webhooks to hold a reference to `_platformFunctions`, I'd like to keep this as a singleton that can be reused for webhooks and across multiple Stripe instances.